reactor-core/src/main/java/reactor/util/context/Context4.java

```diff
@@ -160,6 +160,14 @@ public int size() {
 				new AbstractMap.SimpleImmutableEntry<>(key4, value4));
 	}
 
+	@Override
+	protected AbstractContext putAllSelfInto(AbstractContext initial) {
```

Comment:
It looks like if we change the signature of this method to `Context  putAllSelfInto(Context initial)` the implementation will be the same, which makes me think that we're missing some internal optimizations we can apply (especially when `initial` is `ContextN`)

Reply:
indeed. that would eliminate the need to override `put` return type from `Context` to `AbstractContext`. I'm not sure it means much more than that in terms of further optimizations.

Reply:
I'll implement a special case that creates a new `ContextN` and directly add further values to its internal `Map`, for `Context``1-5`.